scripts/pre-commit

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+echo "********************************"
+echo "Checking code formatting"
+echo "********************************"
+
+ktlint -F --android app/src/**/*.kt data/src/**/*.kt domain/src/**/*.kt testing/src/**/*.kt utility/src/**/*.kt
```

Comment:
@BenHenning do you want me to add a comment here, what does the `-F` is doing in particular?

Reply:
Hmm. This actually wasn't what I had in mind. I was thinking that the pre-commit hook would *fail* if there are linter issues, and during failure would print out this line (with the ``-F``) so that the user could use it to fix most of the issues. I don't think we should auto-fix linter issues (this falls under the general idea that we should never automatically change people's code without their explicit request).

@vinitamurthi @rt4914 any thoughts?

Reply:
I agree with not modifying without the user's explicit approval. So we should run ktlint without -F , mention the errors, and then mention how to fix them (i.e. use the -F flag  as Ben mentioned)

Reply:
I agree that we should not change code automatically as personally I think that can lead to bad experience especially new comers.